assets/js/stripe-payment-request.js

```diff
@@ -402,6 +402,19 @@ jQuery( function( $ ) {
 							} );
 						} );
 
+						$( window ).on( 'load', function() {
```

Comment:
@hardipparmar-multidots could you please explain why is it necessary to run this function on load? As I can see does the same as `$( document.body ).on( 'woocommerce_variation_has_changed updated_addons', function () { ... })` a few lines above.

Reply:
This code needed when someone has already put the attribute product in the cart and they again click that attribute product into the cart page OR someone has open the product using `?attributes`. At that time the first time, we should check the attribute in on load using URL so stripe payment should be shown based on URL attributes.

Reply:
I found an alternative solution, I implemented it [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1149/files#diff-5e2a1337dc63b869d573dcd12e633c5fR244). My solution is more efficient because it puts the correct variation data in the page, instead of requiring an extra AJAX request.

Reply:
Than Good, 
for a quick solution, I have implemented this solution.

But if your solution is more efficient so it's good for us.

Thanks.